pkg/relayer/cmd/cmd.go

```diff
@@ -0,0 +1,250 @@
+	// Set --node flag to the --sequencer-node for the client context
+	cmd.Flags().Set(cosmosflags.FlagNode, fmt.Sprintf("tcp://%s", sequencerNode))
+
+	nodeURL, err := cmd.Flags().GetString(cosmosflags.FlagNode)
+	if err != nil {
+		return nil, err
+	}
+
+	deps, err = supplyMiner(deps)
+	if err != nil {
+		return nil, err
+	}
+
+	deps = supplyEventsQueryClient(deps, nodeURL)
+
+	deps, err = supplyBlockClient(ctx, deps, nodeURL)
+	if err != nil {
+		return nil, err
+	}
+
+	deps, err = supplyTxClient(ctx, deps, cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	deps, err = supplySupplierClient(deps)
+	if err != nil {
+		return nil, err
+	}
+
+	deps, err = supplyRelayerProxy(deps)
+	if err != nil {
+		return nil, err
+	}
+
+	deps, err = supplyRelayMiner(ctx, deps)
+	if err != nil {
+		return nil, err
+	}
+
+	return deps, nil
+}
+
+func supplyMiner(
+	deps depinject.Config,
+) (depinject.Config, error) {
+	mnr, err := miner.NewMiner()
+	if err != nil {
+		return nil, err
+	}
+
+	return depinject.Configs(deps, depinject.Supply(mnr)), nil
+}
+
+func supplyEventsQueryClient(deps depinject.Config, nodeURL string) depinject.Config {
+	eventsQueryClient := eventsquery.NewEventsQueryClient(nodeURL)
+
+	return depinject.Configs(deps, depinject.Supply(eventsQueryClient))
+}
+
+func supplyBlockClient(
+	ctx context.Context,
+	deps depinject.Config,
+	nodeURL string,
+) (depinject.Config, error) {
+	blockClient, err := block.NewBlockClient(ctx, deps, nodeURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return depinject.Configs(deps, depinject.Supply(blockClient)), nil
+}
+
+func supplyTxClient(
+	ctx context.Context,
+	deps depinject.Config,
+	cmd *cobra.Command,
+) (depinject.Config, error) {
+	clientCtx, err := cosmosclient.GetClientTxContext(cmd)
+	if err != nil {
+		return nil, err
+	}
+	clientFactory, err := cosmostx.NewFactoryCLI(clientCtx, cmd.Flags())
+	if err != nil {
+		return nil, err
+	}
+
+	deps = depinject.Configs(deps, depinject.Supply(clientCtx, clientFactory))
+	txContext, err := tx.NewTxContext(deps)
+	if err != nil {
+		return nil, err
+	}
+
+	deps = depinject.Configs(deps, depinject.Supply(txContext))
+	txClient, err := tx.NewTxClient(
+		ctx,
+		deps,
+		tx.WithSigningKeyName(signingKeyName),
+		// TODO_TECHDEBT: populate this from some config.
+		tx.WithCommitTimeoutBlocks(tx.DefaultCommitTimeoutHeightOffset),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return depinject.Configs(deps, depinject.Supply(txClient)), nil
+}
+
+func supplySupplierClient(deps depinject.Config) (depinject.Config, error) {
+	supplierClient, err := supplier.NewSupplierClient(
+		deps,
+		supplier.WithSigningKeyName(signingKeyName),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return depinject.Configs(deps, depinject.Supply(supplierClient)), nil
+}
+
+func supplyRelayerProxy(deps depinject.Config) (depinject.Config, error) {
+	// TODO_INCOMPLETE: this should be populated from some relayerProxy config.
+	anvilURL, err := url.Parse("ws://anvil:8547/")
```

Comment:
1. Move into a local constna
2. Change to TODO_TECHDEBT
3. This shold come form on-chain data

Reply:
This is not what's coming from on-chain data. This is the local node the relays miner is proxying requests to